sql/core/src/test/resources/sql-tests/inputs/group-by-ordinal.sql

```diff
@@ -52,8 +52,19 @@ select count(a), a from (select 1 as a) tmp group by 2 having a > 0;
 -- mixed cases: group-by ordinals and aliases
 select a, a AS k, count(b) from data group by k, 1;
 
--- turn of group by ordinal
+-- turn off group by ordinal
 set spark.sql.groupByOrdinal=false;
 
 -- can now group by negative literal
 select sum(b) from data group by -1;
+
+-- SPARK-21580 ints in aggregation expressions are taken as group-by ordinal
+select 4, b from data group by 1, 2;
+
+set spark.sql.groupByOrdinal=true;
+
+select 4, b from data group by 1, 2;
+
+select 3, 4, sum(b) from data group by 1, 2;
```

Comment:
Only the methods like `Dataset.show` causes en-entrance of analyzed plans. `collect` won't. I guess the queries here are evaluated with `collect`.

    scala> sql("select 4, b, sum(b) from data group by 1, 2").show
    org.apache.spark.sql.AnalysisException: GROUP BY position 4 is not in select list (valid range is [1, 3]); line 1 pos 7

    scala> sql("select 4, b, sum(b) from data group by 1, 2").collect
    res2: Array[org.apache.spark.sql.Row] = Array([4,3,3], [4,4,4], [4,2,2])